archunit/src/main/java/com/tngtech/archunit/lang/EvaluationResult.java

```diff
@@ -45,29 +54,32 @@
  * </code></pre>
  */
 public final class EvaluationResult {
+    static final String ARCHUNIT_IGNORE_PATTERNS_FILE_NAME = "archunit_ignore_patterns.txt";
+    private static final String COMMENT_LINE_PREFIX = "#";
+
     private final HasDescription rule;
     private final List<ConditionEvent> violations;
     private final Optional<String> informationAboutNumberOfViolations;
     private final Priority priority;
 
     @PublicAPI(usage = ACCESS)
     public EvaluationResult(HasDescription rule, Priority priority) {
-        this(rule, emptyList(), Optional.empty(), priority);
+        this(rule, new ArrayList<>(), Optional.empty(), priority);
     }
 
     @PublicAPI(usage = ACCESS)
     public EvaluationResult(HasDescription rule, ConditionEvents events, Priority priority) {
         this(
                 rule,
-                events.getViolating(),
+                new ArrayList<>(events.getViolating()),
                 events.getInformationAboutNumberOfViolations(),
                 priority
         );
     }
 
-    EvaluationResult(HasDescription rule, Collection<ConditionEvent> violations, Optional<String> informationAboutNumberOfViolations, Priority priority) {
+    private EvaluationResult(HasDescription rule, ArrayList<ConditionEvent> violations, Optional<String> informationAboutNumberOfViolations, Priority priority) {
```

Comment:
I wanted to avoid copying lists around internally. `EvaluationResult.add(..)` mutates the list, so we need to ensure it's a writable list. And from that point of view I thought better make it explicit internally then :man_shrugging: Otherwise we could also drop that and always safe-copy the list, etc., but in this place we're potentially dealing with a big number of elements...
Do you think this is prematurely optimizing? I have to admit that I didn't run any explicit tests here, it's more a gut feeling, and it might be wrong now that we only store violations here, instead of all events as before the memory optimization.

Reply:
Okay, I see. Thanks! So it was basically an unrelated bug that `emptyList()` from the previous `EvaluationResult(HasDescription, Priority)` constructor did not support mutation? But anyways; for the ultimate consistency, should we change `EvaluationResult`'s field `violations` to a (mutable) `ArrayList`, too?

Reply:
Yeah, would likely be a good idea. I think the problem with this (and the "you should use interfaces" best practice) is that the Java `Collection` hierarchy was designed sub-optimally. Because `List` suggests it is a mutable collection, but there are too many implementations (and in particular `ImmutableList` which we use heavily) that do not fulfill this contract.